setup.py

```diff
@@ -17,20 +17,19 @@
 # IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
 # WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
-import setuptools
```

Comment:
setuptools should always be imported first because it does some monkey patching of distutils.

We should add a comment to indicate that.

Reply:
distutils is going away; see https://github.com/jbms/sphinx-immaterial/pull/45#issuecomment-1072288302

until we phase out the use of distutils, I could add a pylint comment to ignore the import ordering, here.

Reply:
I think distutils isn't exactly going away, it is just being moved out of the Python standard library into setuptools itself.

Furthermore, I think setuptools modifies `sys.modules` so that you can still import distutils using the normal path:
https://github.com/pypa/setuptools/blob/main/_distutils_hack/__init__.py

Does the deprecation warning still appear if you fix the import order?

Reply:
pylint realizes the setuptools is not in std lib, so it complains about the order. Using
```python

import setuptools  # pylint: disable=wrong-import-order

import atexit
import distutils.command.build
import os
import subprocess
import tempfile

import setuptools.command.build_py
import setuptools.command.develop
import setuptools.command.install
import setuptools.command.sdist
```
doesn't seem to trigger the deprecatin warning. 🤷🏼

Reply:
After reading [backward compatibility in PEP 632](https://peps.python.org/pep-0632/#backwards-compatibility), I found that `setuptools` (as of v60.0) has import paths that help mitigate the removal of distutils. According to [the setuptools docs](https://setuptools.pypa.io/en/latest/deprecated/distutils-legacy.html#prefer-setuptools), we should use 
```ini
# pyproject.toml
[build-system]
requires = ["setuptools>=60", "wheel", "setuptools_scm"]
```
```python
# setup.py
try:
    import setuptools.command.build
except ImportError:  # for setuptools<60.0
    import distutils.command.build
```

Reply:
I don't think `setuptools.command.build` is present in setuptools 60.  It isn't listed on that webpage and I tried installing setuptools 60.10.0 and get an import error.

However, as long as we import setuptools first we are in fact using the version of distutils from setuptools, which is why there is no deprecation warning.

Reply:
Strange, that deprecation warning is completely useless then. All it did was cause confusion here.